### ISSUE TITLE
Crash in TVIEW.BAS

### ISSUE THREAD

**Issue by robhagemans** [robhagemans/pcbasic#42]
https://sourceforge.net/p/pcbasic/discussion/general/thread/6270c85d/
```

10:45:52 rob@bandicoot:~/Downloads/TVIEW_85.ZIP.1$ ~/Projects/basic-project/pc-basic/pcbasic.py -l=TVIEW.BAS --preset=pcjr
ERROR: Traceback (most recent call last):
  File "/home/rob/Projects/basic-project/pc-basic/pcbasic/basic/session.py", line 381, in _handle_exceptions
    yield
  File "/home/rob/Projects/basic-project/pc-basic/pcbasic/basic/session.py", line 291, in interact
    self.interpreter.loop()
  File "/home/rob/Projects/basic-project/pc-basic/pcbasic/basic/interpreter.py", line 107, in loop
    self.parse()
  File "/home/rob/Projects/basic-project/pc-basic/pcbasic/basic/interpreter.py", line 96, in parse
    self.parser.parse_statement(ins)
  File "/home/rob/Projects/basic-project/pc-basic/pcbasic/basic/parser/statements.py", line 90, in parse_statement
    self._callbacks[c](parse_args(ins))
  File "/home/rob/Projects/basic-project/pc-basic/pcbasic/basic/display.py", line 681, in screen_
    self.screen(mode, colorswitch, apagenum, vpagenum, erase)
  File "/home/rob/Projects/basic-project/pc-basic/pcbasic/basic/display.py", line 755, in screen
    self.mode.video_segment*0x10, self.video_mem_size)
  File "/home/rob/Projects/basic-project/pc-basic/pcbasic/basic/modes.py", line 311, in get_memory
    bytes = [0]*num_bytes
TypeError: can't multiply sequence by non-int of type 'float'


10:46:48 rob@bandicoot:~/Downloads/TVIEW_85.ZIP.1$ ~/Projects/basic-project/branch1508/pcbasic.py -l=TVIEW.BAS --preset=pcjr
ERROR: Traceback (most recent call last):
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/run.py", line 66, in run_once
    if not statements.parse_statement():
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/statements.py", line 169, in parse_statement
    elif c == tk.SCREEN:    exec_screen(ins)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/statements.py", line 2501, in exec_screen
    screen.screen(mode, color, apagenum, vpagenum, erase)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/backend.py", line 1026, in screen
    self.mode.set_memory(self.mode.video_segment*0x10, save_mem)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/modes.py", line 688, in set_memory
    bytes_to_interval(bytes[ofs:ofs+length], self.ppb))
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/backend.py", line 1355, in put_interval
    video.put_interval(pagenum, x, y, colours, mask)
  File "/home/rob/Projects/basic-project/branch1508/pcbasic/video_pygame.py", line 1272, in put_interval
    ref |= colours
TypeError: ufunc 'bitwise_or' output (typecode 'l') could not be coerced to provided output parameter (typecode 'B') according to the casting rule ''same_kind''


```

**Comment by robhagemans** [comment]
15.08 error fixed by 60b0199

**Comment by robhagemans** [comment]
16.12 fixed by 894cbb9
